### PR TITLE
Added API specifiaction

### PR DESCRIPTION
**Pull request by mtakac2**
First draft of API specs. This should be enough to help us move `/submissions/:id` page to the new client.

### CODE REVIEW

File: API_SPEC.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
+1. [Submissions](#submissions)
+    1. [GET /submissions/:id](#get-submissionsid)
+    2. [PUT /submissions/:id](#put-submissionsid)
```
Comment (kytrinyx) [inline API_SPEC.md]: Should this be a PATCH?

File: API_SPEC.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+Gets submission object
+
+#### Responses
+**200**:
+```
+{
+    "_link": "/submissions/d5f76a20f4b847c59d2d9b65f134c9ef",
+    "views": "4",
+    "solution": "Solution code here...",
+    "acceptsFeedback": "true",
```
Comment (kytrinyx) [inline API_SPEC.md]: The current behavior is that people can always post feedback, no matter what state of the exercise is ("pending", "hibernating", "done").

File: API_SPEC.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    "body": "Is there an advantage to write def `self.compute` instead if def `Hamming.compute`?",
+    "createdAt": "Datetime in some form",
+    "user": {
+        "_link": "/agreen",
+        "username": "agreen"
+    }
+}
+```
+
+**404**
```
Comment (kytrinyx) [inline API_SPEC.md]: I imagine that we would have a json response for the 404 here (as above in GET /submissions/:uuid)

File: API_SPEC.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    "bodyRaw": "Lorem ipsum dolor sit amet"
+}
+
+#### Responses
+**201**:
+```
+{
+    "_link": "/comments/2",
+    "bodyRaw": "Lorem ipsum dolor sit amet",
+    "body": "Lorem ipsum dolor sit amet",
```
Comment (kytrinyx) [inline API_SPEC.md]: The bodyRaw is markdown, whereas the body is html, correct?

File: API_SPEC.md (hunk 3 above)
Comment (kytrinyx) [inline API_SPEC.md]: Update -- oh, yeah, I see now. This is omitting the obvious stuff. Good call.

**Comment by kytrinyx** [comment]
This looks good. I just have the one question about the acceptingFeedback attribute.

File: API_SPEC.md (hunk 1 above)
Comment (mtakac) [inline API_SPEC.md]: Yes, my mistake.

File: API_SPEC.md (hunk 2 above)
Comment (mtakac) [inline API_SPEC.md]: Ok, in that case `state` attribute wil be probably better.

File: API_SPEC.md (hunk 3 above)
Comment (mtakac) [inline API_SPEC.md]: Yes, I am looking into a way how could swagger or something similar let us define stuff like this just once and then reuse it. I'll update it shortly.

File: API_SPEC.md (hunk 4 above)
Comment (mtakac) [inline API_SPEC.md]: Correct. Maybe they could be named better. I just didn't want to fixate to the Markdown. I know it's unlikely, but if the parser would ever change, we would have to update the API if markdown would be part of the name of the attribute.

File: API_SPEC.md (hunk 2 above)
Comment (kytrinyx) [inline API_SPEC.md]: Yeah, that sounds good.

File: API_SPEC.md (hunk 4 above)
Comment (kytrinyx) [inline API_SPEC.md]: Yeah, definitely. I think bodyRaw and body are good.

**Comment by kytrinyx** [comment]
I was entirely overconfident in my ability to spend time on a new API—this has just not been possible in the past year.

I have opened a new issue concerning a full rewrite of the entire site, which would include rethinking features and APIs and interactions with the CLI.

This lives in the following issue: https://github.com/exercism/discussions/issues/29 I'm going to close this issue.

Thanks for putting so much thought into this!

**Pull request closed without merging**